applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -145,28 +144,57 @@ static void handle_dvfs_error(int32_t err)
 	(void) k_work_reschedule(&dvfs_retry.retry_work, timeout);
 }
 
-static void set_dvfs_freq(enum dvfs_frequency_setting target_freq)
+static void dvfs_notify_cb(void)
+{
+	int res;
+	int32_t ret = sys_notify_fetch_result(&cli.notify, &res);
+
+	if (ret < 0) {
+		LOG_DBG("work not completed");
+		return;
+	}
+
+	ret = nrf_clock_control_release(dvfs_clock_dev, &spec);
```

Comment:
Discussed offline. This seems wrong use of API. The given request should be kept active until the frequency is no longer needed.

Reply:
We could keep a request per bit in dvfs state bitmask or rely on only 1 request common for all of the bits (preferrable the second approach - lower memory consumption, simpler change).